crates/gpui/src/platform/windows/events.rs

```diff
@@ -273,7 +274,8 @@ fn handle_syskeydown_msg(
         keystroke,
         is_held: lparam.0 & (0x1 << 30) > 0,
     };
-    let result = if func(PlatformInput::KeyDown(event)).default_prevented {
+    let result = if !func(PlatformInput::KeyDown(event)).propagate {
+        state_ptr.state.borrow_mut().system_key_handled = true;
```

Comment:
I'm not sure how heavy `RefCell` is. Should I use `AtomicBool` for this case instead?

Reply:
RefCell should be a lot cheaper, as there's no atomic synchronization.